dex/farm/tests/farm_setup/multi_user_farm_setup.rs

```diff
@@ -472,6 +492,51 @@ where
             .assert_ok();
     }
 
+    pub fn update_energy_for_user(&mut self, error: bool) {
+        let b_mock = &mut self.b_mock;
+        let user_addr = &self.first_user;
+
+        if error {
+            b_mock
+                .execute_tx(
+                    &self.first_user,
+                    &self.farm_wrapper,
+                    &rust_biguint!(0),
+                    |sc| {
+                        sc.update_energy_for_user(managed_address!(user_addr));
+                    },
+                )
+                .assert_error(4, "Can only manually update energy only after claim")
+        } else {
+            b_mock
+                .execute_tx(
+                    &self.first_user,
+                    &self.farm_wrapper,
+                    &rust_biguint!(0),
+                    |sc| {
+                        sc.update_energy_for_user(managed_address!(user_addr));
+                    },
+                )
+                .assert_ok();
+        }
```

Comment:
Can be rewritten as:
let result = b_mock......;
if error {
    result.assert_error();
} else { result.assert_ok();